### PR TITLE
global: fix componentDidMount and add links to next/previous journal in series

### LINKED ISSUE #769: add next/previous journal on the series details pages
If the series object has sequence relation (previous or next) 
- [x] display them on the series details page of the patrons view(frontsite)
- [x] display this information in the metadata of the series details page on the backoffice 
before you start please propose the placement for this information.
The goal is to communicate to the librarian/patron that the current series `is continued by` another one or `is a continuation` of a series (accordingly to next/previous relation)

### PR DESCRIPTION
**Pull request by KonstantinaStoikou**
Fix componentDidMount in details js files (because links on details pages didn't work) and add links to next/previous journal in series.
Closes #769.

Selection_017

### CODE REVIEW

File: ui/src/pages/backoffice/Acquisition/Vendor/VendorDetails/VendorDetails.js [hunk 1]
```diff
@@ -150,16 +149,13 @@ export default class VendorDetails extends React.Component {
...
-        this.props.fetchVendorDetails(loc.state.pid);
-      }
-    });
     this.props.fetchVendorDetails(this.props.match.params.vendorPid);
   }
 
-  componentWillUnmount() {
-    this.unlisten();
+  componentDidUpdate(prevProps) {
+    if (prevProps.match.params.orderPid !== this.props.match.params.vendorPid) {
```
Comment (ntarocco) [inline ui/src/pages/backoffice/Acquisition/Vendor/VendorDetails/VendorDetails.js]:
```suggestion
    if (prevProps.match.params.vendorPid !== this.props.match.params.vendorPid) {
```

File: ui/src/pages/backoffice/Acquisition/Order/OrderDetails/OrderDetails.js [hunk 2]
```diff
@@ -208,16 +207,13 @@ export default class OrderDetails extends React.Component {
...
-        this.props.fetchOrderDetails(loc.state.pid);
-      }
-    });
     this.props.fetchOrderDetails(this.props.match.params.orderPid);
   }
 
-  componentWillUnmount() {
-    this.unlisten();
+  componentDidUpdate(prevProps) {
+    if (prevProps.match.params.orderPid !== this.props.match.params.orderPid) {
```
Comment (ntarocco) [inline ui/src/pages/backoffice/Acquisition/Order/OrderDetails/OrderDetails.js]: General very minor comment (don't need to change everything!) for all these changes, to take into account in future code.
To keep the code more readable given the Prettier reformatting, I would have done for all of them:
```
const orderPid = this.props.match.params.orderPid
const samePidFromRouter = prevProps.match.params.orderPid === orderPid
if (!samePidFromRouter) {
    this.props.fetchOrderDetails(orderPid)
}
```
This is to remember that is coming from the Router and why we have that.
@KonstantinaStoikou WDYT?

**Review by ntarocco — COMMENTED** [review]
Thanks a lot for this work, it was important to fix it once and for all! :)
I left one comment to be fixed (wrong copy/paste)

👏

File: ui/src/pages/backoffice/Acquisition/Order/OrderDetails/OrderDetails.js (hunk 2 above)
Comment (KonstantinaStoikou) [reply ui/src/pages/backoffice/Acquisition/Order/OrderDetails/OrderDetails.js]: I agree with what you propose. Do you want me to change it?

**Review by kpsherva — APPROVED** [review]

**Review by ntarocco — APPROVED** [review]

**Pull request merged**